app/src/main/java/com/example/taskwise_eventmaster/presentation/events_screen/EventsViewModel.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.taskwise_eventmaster.presentation.events_screen
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.taskwise_eventmaster.domain.adapters.eventToTask
+import com.example.taskwise_eventmaster.domain.repository.EventRepository
+import com.example.taskwise_eventmaster.domain.repository.TaskRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class EventsViewModel @Inject constructor(
+    private val repositoryEvent: EventRepository,
+    private val repositoryTask: TaskRepository,
+) : ViewModel() {
+    var state by mutableStateOf(EventsState())
+        private set
+
+    init {
+        fetchEventInfo()
+    }
+
+    fun onEvent(event: EventsScreenEvent) {
+        when (event) {
+            is EventsScreenEvent.SaveEventInCalendar -> saveEventInCalendar(event)
+        }
+    }
+
+    private fun saveEventInCalendar(event: EventsScreenEvent.SaveEventInCalendar) {
```

Comment:
If I’ve understood your question correctly, my answer would be that I’m not using 'Event' as a parameter here because I have a class named Event, which should be called by the 'Event' function. To conclude, 'EventEvent' wasn’t appealing to me, so I named it EventScreenEvent.